ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java

```diff
@@ -223,6 +224,46 @@ static String getFullyQualifiedDomainName(String unqualifiedHostname) {
     }
   }
 
+  /**
+   * Validate hostName.
+   * @param clusterMapResolveHostnames indicates if a reverse DNS lookup is enabled or not.
+   * @param hostName hostname to be validated.
+   * @throws IllegalStateException if hostname is not valid.
+   */
+  public static void validateHostName(Boolean clusterMapResolveHostnames, String hostName) {
+    if (clusterMapResolveHostnames) {
+      String fqdn = getFullyQualifiedDomainName(hostName);
+      if (!fqdn.equals(hostName)) {
+        throw new IllegalStateException(
+            "Hostname for AmbryDataNode (" + hostName + ") does not match its fully qualified domain name: " + fqdn);
+      }
+    }
+  }
+
+  /**
+   * Validate plainTextPort and sslPort.
+   * @param plainTextPort PlainText {@link Port}.
+   * @param sslPort SSL {@link Port}.
+   * @throws IllegalStateException if ports are not valid.
+   */
+  public static void validatePort(Port plainTextPort, Port sslPort, boolean sslPortRequired) {
+    if (sslPort != null) {
+      if (sslPort.getPort() == plainTextPort.getPort()) {
+        throw new IllegalStateException("Same port number for both plain and ssl ports");
+      }
+      if (sslPort.getPort() < MIN_PORT || sslPort.getPort() > MAX_PORT) {
+        throw new IllegalStateException(
+            "SSL Port " + plainTextPort.getPort() + " not in valid range [" + MIN_PORT + " - " + MAX_PORT + "]");
+      }
+    } else if (sslPortRequired) {
+      throw new IllegalArgumentException("No SSL port to a datanode to which SSL is enabled.");
+    }
+    if (plainTextPort.getPort() < MIN_PORT || plainTextPort.getPort() > MAX_PORT) {
+      throw new IllegalStateException(
```

Comment:
Minor: I realize this is copied code, but it seems strange to throw IllegalArgumentException for null value and IllegalStateException for invalid value.  Seems like we can use IllegalArgument for both cases (and update the javadoc).